src/test/java/com/airlines/dao/impl/CrewDaoImplTest.java

```diff
@@ -0,0 +1,182 @@
+package com.airlines.dao.impl;
+
+import com.airlines.dao.CrewDao;
+import com.airlines.model.Crew;
+import com.airlines.model.CrewMember;
+import com.airlines.util.JdbcUtil;
+import com.airlines.util.TestDatasourceProvider;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.airlines.model.Citizenship.BRAZILIAN;
+import static com.airlines.model.Citizenship.ITALIAN;
+import static com.airlines.model.Position.PILOT_IN_COMMAND;
+import static org.junit.Assert.*;
+
+public class CrewDaoImplTest {
+    private static final String SCHEMA_SQL_PATH = "src/test/resources/sql/schema.sql";
+    private static final String DATA_SQL_PATH = "src/test/resources/sql/test_data.sql";
+    private static final String CREATE_TABLES_QUERY = JdbcUtil.getSqlQueryString(SCHEMA_SQL_PATH);
+    private static final String GENERATE_DATA_QUERY = JdbcUtil.getSqlQueryString(DATA_SQL_PATH);
+
+    private static final CrewMember CREW_MEMBER = buildTestCrewMember();
+    private static final Crew CREW = buildCrewFromDataSet();
+    private static final Crew INVALID_CREW = buildInvalidCrewFromDataSet();
+
+    private static CrewDao crewDao;
+
+    private static DataSource dataSource;
+
+    @BeforeClass
+    public static void init() {
+        dataSource = TestDatasourceProvider.createTestDatasource();
+        crewDao = new CrewDaoImpl(dataSource);
+    }
+
+    @Before
+    public void dataBaseInit() throws SQLException {
+        createTables();
+    }
+
+    @Test
+    public void shouldThrowExceptionIfInvalidArgumentCrewMemberProvidedToAddMethod() {
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> crewDao.add(null, null));
+        assertEquals("Valid crewMember entity should be provided", exception.getMessage());
+    }
+
+    @Test
+    public void shouldThrowExceptionIfInvalidArgumentCrewProvidedToAddMethod() {
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> crewDao.add(CREW_MEMBER, null));
+        assertEquals("Valid crew entity should be provided", exception.getMessage());
+    }
+
+    @Test
+    public void shouldCorrectlyAddCrewMemberToCrew() throws SQLException {
+        populateTestData();
+        crewDao.add(CREW_MEMBER, CREW);
+        List<CrewMember> list = crewDao.getByCrewId(3L);
+        assertTrue(list.contains(CREW_MEMBER));
+    }
+
+    @Test
+    public void shouldGetCrewMembersByCrewId() throws SQLException {
+        populateTestData();
+        List<CrewMember> list = crewDao.getByCrewId(3L);
+        assertEquals(1, list.size());
+    }
+
+    @Test
+    public void shouldThrowExceptionIfCrewIsNotProvidedForGetByIdMethod() throws SQLException {
```

Comment:
Do you really need declaring throwing SQLException here?
IDEA marked it in grey color, so it means it is not used